libs/plugins/stream-client/src/lib/core/stream.service.ts

```diff
@@ -137,4 +144,42 @@ export class StreamService {
         map(() => itemId)
       );
   }
+
+  deleteStream(stream) {
+    return this.translate
+      .get(['STREAMS.DELETE-STREAM:CONFIRM_DELETE', 'MODAL:CONFIRM-DELETION'], {
+        streamName: stream.name,
+      })
+      .pipe(
+        switchMap(translation => {
+          return this.confirmationModal.openModal({
+            title: translation['MODAL:CONFIRM-DELETION'],
+            textMessage: translation['STREAMS.DELETE-STREAM:CONFIRM_DELETE'],
+          }).result;
+        }),
+        filter(result => result === ConfirmationResult.Confirm),
+        switchMap(() => {
+          return this.appResourceService.deleteResourceWithTrigger(stream.id, null).pipe(
```

Comment:
We can update the logic such that we do not have to have an empty subscription in the component. 
```
this.translate
      .get()
      .pipe(
        switchMap(translation => {
          return this.confirmationModal.openModal().result;
        }),
        filter(result => result === confirm),
        switchMap(() => this.appResourceService.deleteResourceWithTrigger(stream.id, null))
      ).subscribe(() => {
          this.navigateToApp(stream.app.id);
          notify user
        }, err => {
          console.error(err);
          this.notifications.error({
            key: 'STREAMS.DELETE-STREAM:ERROR-MESSAGE-DELETE-STREAM',
            params: err,
          });
        });
```